File: Programmers_java/네트워크.java
import java.util.*;

class Solution {
    static boolean[] visited;

    public int solution(int n, int[][] computers) {
        int answer = 0;
        visited = new boolean[computers.length];

        for (int i = 0; i < computers.length; i++) {
            if (!visited[i]) {
                answer++;
                BFS(i, computers);
            }
        }
        return answer;
    }

    public static void BFS(int i, int[][] arr) {
        Queue<Integer> queue = new LinkedList<>();
        queue.add(i);

        while (!queue.isEmpty()) {
            int cur = queue.poll();
            visited[cur] = true;

            for (int j = 0; j < arr.length; j++) {
                if (!visited[j] && arr[cur][j] == 1) {
                    queue.add(j);
                }
            }
        }
    }
}

File: Programmers_python/네트워크.py
from collections import deque

def bfs(computers, visited, start):
    q = deque([start])
    
    while q:
        cur = q.popleft()
        for j in range(len(computers[cur])):
            if computers[cur][j] == 1 and not visited[j]:
                visited[j] = True
                q.append(j)
    
    
def solution(n, computers):
    answer = 0
    visited = [False] * n
    
    for i in range(len(computers)):
        if not visited[i]:
            visited[i] = True
            answer += 1
            bfs(computers, visited, i)
    
    return answer
